core/src/main/scala/org/apache/spark/sql/crossdata/catalog/PersistentCatalog.scala

```diff
@@ -0,0 +1,186 @@
+
+      logInfo(s"PersistentCatalog: Looking up table ${relationIdentifier.unquotedString}")
+
+      lookupTable(relationIdentifier) match {
+        case Some(crossdataTable) =>
+          val table: LogicalPlan = createLogicalRelation(crossdataTable)
+          processAlias(relationIdentifier, table, alias)
+
+        case None =>
+          log.debug(s"Table Not Found: ${relationIdentifier.unquotedString}")
+          lookupView(relationIdentifier) match {
+            case Some(sqlView) =>
+              val viewPlan: LogicalPlan = xdContext.sql(sqlView).logicalPlan
+              processAlias(relationIdentifier, viewPlan, alias)
+            case None =>
+              log.debug(s"View Not Found: ${relationIdentifier.unquotedString}")
+              sys.error(s"Table/View Not Found: ${relationIdentifier.unquotedString}")
+          }
+      }
+
+  }
+
+  override def registerTable(tableIdentifier: TableIdentifier, plan: LogicalPlan): Unit = ()
+  override def unregisterTable(tableIdentifier: TableIdentifier): Unit = ()
+
+  override def registerView(viewIdentifier: ViewIdentifier, plan: LogicalPlan): Unit = ()
+  override def unregisterView(viewIdentifier: ViewIdentifier): Unit = ()
+
+  override def unregisterAllTables(): Unit = ()
+  override def unregisterAllViews(): Unit = ()
+
+  override def refreshTable(tableIdentifier: TableIdentifier): Unit = {
+    throw new UnsupportedOperationException
+  }
+
+  final override def persistTable(crossdataTable: CrossdataTable, table: LogicalPlan): Unit = {
+    val tableIdentifier = TableIdentifier(crossdataTable.tableName, crossdataTable.dbName)
+
+    if (tableExists(tableIdentifier)){
+      logWarning(s"The table $tableIdentifier already exists")
+      throw new UnsupportedOperationException(s"The table $tableIdentifier already exists")
+    } else {
+      logInfo(s"Persisting table ${crossdataTable.tableName}")
+      persistTableMetadata(crossdataTable.copy(schema = Option(table.schema)))
+    }
+  }
+
+  final override def persistView(tableIdentifier: ViewIdentifier, plan: LogicalPlan, sqlText: String) = {
+    def checkPlan(plan: LogicalPlan): Unit = {
+      plan collect {
+        case UnresolvedRelation(tIdent, _) => tIdent
+      } foreach { tIdent =>
+        if (!tableExists(tIdent)) {
+          throw new RuntimeException("Views only can be created with a previously persisted table")
+        }
+      }
+    }
+    checkPlan(plan)
+    if (tableExists(tableIdentifier)){
+      val msg = s"The view ${tableIdentifier.unquotedString} already exists"
+      logWarning(msg)
+      throw new UnsupportedOperationException(msg)
+    } else {
+      logInfo(s"Persisting view ${tableIdentifier.unquotedString}")
+      persistViewMetadata(tableIdentifier: TableIdentifier, sqlText)
+    }
+  }
+
+}
+
+object PersistentCatalog  extends CrossdataSerializer {
+  def getUserSpecifiedSchema(schemaJSON: String): Option[StructType] = {
```

Comment:
This refactor could be used to replace the way we did the serialization and take advantage of the CrossdataSerializer